README
Dist::Zilla::Plugin::Bootstrap::ShareDir::Dist

This module allows one to load a "Dist" styled "ShareDir" using a
"Bootstrap" mechanism so a distribution can use files in its own source
tree when building with itself.

This is very much like the "Bootstrap::lib" plugin in that it injects
libraries into @INC based on your existing source tree, or a previous
build you ran.

And it is syntactically like the "ShareDir" plugin.

Note that this is really only useful for self consuming *plugins* and
will have no effect on the "test" or "run" phases of your dist. ( For
that, you'll need "Test::File::ShareDir" ).

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-Plugin-Bootstrap-ShareDir-Dist

1.001003 2017-03-04T07:06:18Z 788efd9

1.001002 2017-03-04T06:54:21Z 8830111
 [Bugfix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Plugin-Bootstrap-ShareDir-Dist/issues/3

 [Dependencies::Stats]
 - Dependencies changed since 1.001001, see misc/*.deps* for details
 - develop: +5 ↑4 -2 (suggests: ↑2)
 - runtime: -1
 - test: ↓1

 [Internals]
 - Drop ConfigDumper

1.001001 2015-01-20T22:33:27Z d97fa56
 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +10 ↑2 -2 (recommends: +1, suggests: ↑2)
 - runtime: -1 (recommends: +1)
 - test: +1 (recommends: +1 ↑1 ↓1)

 [Documentation]
 - Improved DESCRIPTION section added for README.

 [Internals]
 - Removed MooseX::AttributeShortcuts to remove indirect autobox dep.

1.001000 2014-08-30T13:47:50Z
 [00 Minor]
 - Tests refactored.
 - metadata
 - internals

 [Dependencies::Stats]
 - Dependencies changed since 1.000002, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - runtime: +1
 - test: +1 -2

 [Internals]
 - dump_config implemented in config_dumper

 [Metadata]
 - set author = kentnl

 [Tests]
 - Refactored in terms of DZPUtAKENTNL::dztest

1.000002 2014-08-15T21:15:41Z
 [00 Trivial]
 - CPANDAY!
 - no code change.
 - whitespace adjustment.

 [Dependencies::Stats]
 - Dependencies changed since 1.000001, see misc/*.deps* for details
 - develop: +1 (suggests: ↑1)

 [Misc]
 - Improve source side POD to reduce build side whitespace.

1.000001 2014-08-03T02:04:39Z
 [00 Tests]
 - Drop use of Path::FindDev for tests.

 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - test: -2

1.000000 2014-08-02T07:48:52Z
 [00 Major]
 - Changes that may affect hand-operated downstreams.

 [01 Version Scheme]
 - Moved to x.yyyyyy
 - This is the primary reason for the 1.x on the box.
 - https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [02 Toolkit]
 - Moved to ExtUtils::MakeMaker
 - "Latest" dependencies now only a recommendation.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.3.0, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 -11 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +2 ↓1 (recommends: +3)

0.3.0 2013-11-21T07:19:14Z
 [00 Major]
 - Internals refactored in terms of T:F:SD:O:D

 [Dependencies::Stats]
 - Dependencies changed since 0.2.4, see misc/*.deps* for details
 - runtime: +1 -1

0.2.4 2013-11-20T11:51:25Z
 [Bugfix]
 - Drop dependency on Perl 5.10

 [Dependencies::Stats]
 - Dependencies changed since 0.2.3, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +43 ↑1 -1 (recommends: -1, suggests: +1 -1)
 - runtime: ↓1
 - test: ↑1 -3

 [Documentation]
 - Point README to metacpan

 [Packaging]
 - use new merged prereqs stuff

 [Tests]
 - Use Test::Compile::PerFile
 - Report less things with RVT

0.2.3 2013-09-24T22:31:06Z
 [Bugfix]
 - Call Test::DZil instead of calling dzil directly.

 [Dependencies::Stats]
 - Dependencies changed since 0.2.2, see misc/*.deps* for details
 - test: +3

0.2.2 2013-09-08T08:19:00Z
 [Bugfix]
 - Tests require DZP:BS:Lib

 [Dependencies::Stats]
 - Dependencies changed since 0.2.1, see misc/*.deps* for details
 - test: +1

0.2.1 2013-09-04T13:18:15Z
 [!Minor]
 - Make immutable.
 - Tidy

0.2.0 2013-09-04T12:57:46Z
 [!Major]
 - Reimplemented in terms of ::Role::Bootstrap

 [Dependencies::Stats]
 - Dependencies changed since 0.1.0, see misc/*.deps* for details
 - runtime: +1 -3
 - test: +2

0.1.0 2013-09-03T06:40:40Z
 - First version.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Use a share directory on your dist during bootstrap",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Bootstrap-ShareDir-Dist",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "Dist::Zilla::Plugin::Bootstrap::ShareDir::Dist",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::Bootstrap" => 0,
    "Moose" => 0,
    "Test::File::ShareDir::Object::Dist" => 0,
    "lib" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
    "Dist::Zilla::Util::Test::KENTNL" => "1.003001",
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.001003",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
  "Dist::Zilla::Role::Bootstrap" => 0,
  "Dist::Zilla::Util::Test::KENTNL" => "1.003001",
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::File::ShareDir::Object::Dist" => 0,
  "Test::More" => 0,
  "lib" => 0,
  "strict" => 0,
  "utf8" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4206"
  },
  "configure_requires" => {
    "Module::Build" => "0.4206"
  },
  "dist_abstract" => "Use a C<share> directory on your dist during bootstrap",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Bootstrap-ShareDir-Dist",
  "dist_version" => "1.000000",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Bootstrap::ShareDir::Dist",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Role::Bootstrap" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "Test::File::ShareDir::Object::Dist" => 0,
    "lib" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Cwd" => 0,
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
    "ExtUtils::MakeMaker" => 0,
    "File::Copy::Recursive" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "Path::FindDev" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "Cwd" => 0,
  "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
  "ExtUtils::MakeMaker" => 0,
  "File::Copy::Recursive" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "Module::Build" => "0.4206",
  "Path::FindDev" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
